### ISSUE TITLE
Azure CLI bot miss registration to winget-pkg repository

### ISSUE THREAD

**Issue by kkamegawa** [Azure/azure-cli#27492]
### Describe the bug

Since 2.50 a 64bit version of Azure cli has been released, which is very good news for me as I have been waiting for the 64bit version.

However, the azure-cli team has registered wrong information in the winget-pkg repository, so the 32-bit version of the CLI is installed even on 64-bit OS.

Users are unable to properly install the 64-bit version of the azure cli. 2.53 and the upcoming release of the manifest, can you please fix this?

https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/AzureCLI/2.53.0/Microsoft.AzureCLI.installer.yaml

miss manifest(current)
```
- Architecture: **x64**
  InstallerType: msi
  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.53.0.msi
```
correct manifest
```
- Architecture: **x86**
  InstallerType: msi
  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.53.0.msi

- Architecture: **x64**
  InstallerType: msi
  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.53.0**-x64**.msi
```

### Related command

winget install microsoft.azurecli

### Errors

x64 version of Windows install x86 version of Azure CLI.

### Issue script & Debug output

install x86 version of Azure CLI

### Expected behavior

Install x64 version of Azure CLI in Windows x64 with `winget install microsoft.azurecli`.

### Environment Summary

winget upgrade microsoft.azurecli, but can't upgrade x64 version of Azure CLI.
----------------------------------
azure-cli                         2.52.0 *

core                              2.52.0 *
telemetry                          1.1.0

Extensions:
azure-devops                      0.25.0
containerapp                       0.3.9

Dependencies:
msal                            1.24.0b1
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe'

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by horihiro** [comment]
Please look into this issue.

This issue causes the following status.
![image](https://github.com/Azure/azure-cli/assets/4566555/546c79b4-f3e9-4572-a72a-000913a2d890)

As of 10/25/2023, the following steps can reproduce because [the manifest created by azclibot](https://github.com/microsoft/winget-pkgs/blob/5678afb7f8264f3d40dd4cce09364584956dfd9b/manifests/m/Microsoft/AzureCLI/2.53.1/Microsoft.AzureCLI.installer.yaml) tries upgrades to x86 version.

```
winget install --id Microsoft.AzureCLI --version 2.53.0
winget upgrade --id Microsoft.AzureCLI
```

**Comment by bebound** [comment]
@horihiro Please see https://github.com/microsoft/winget-pkgs/pull/123641#issuecomment-1778089246

**Comment by horihiro** [comment]
@bebound Which comment?
I cannot see the comment you refer ( `#issuecomment-1778089246` )

**Comment by jiasli** [comment]
@horihiro, the reason why you are seeing 2 Azure CLIs installed is because you install 2.53.0 64-bit first, then install 2.53.1 32-bit (due to the incorrect manifest). This is explained in https://github.com/Azure/azure-cli/pull/26640#issuecomment-1645337341:

> Installing **32-bit MSI on top of 64-bit MSI** will keep both MSI's side-by-side.

You may delete the old versions manually. With the `InstallerUrl` being fixed for 2.54.0 (https://github.com/microsoft/winget-pkgs/pull/126489), you shouldn't see this issue anymore.

**Comment by jiasli** [comment]
We have fixed the manifest generation script. This is the new template we use to generate `Microsoft.AzureCLI.installer.yaml`:

```sh
cat >Microsoft.AzureCLI.installer.yaml <<EOL
# Created using wingetcreate 1.1.2.0
# yaml-language-server: \$schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json

PackageIdentifier: Microsoft.AzureCLI
PackageVersion: ${CLI_VERSION}
Installers:
- Architecture: x86
  InstallerType: msi
  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-${CLI_VERSION}.msi
  InstallerSha256: ${msi_x86_sum}
- Architecture: x64
  InstallerType: msi
  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-${CLI_VERSION}-x64.msi
  InstallerSha256: ${msi_x64_sum}
ManifestType: installer
ManifestVersion: 1.2.0
EOL
```

BTW, https://github.com/microsoft/winget-pkgs/pull/119228 updated `ManifestVersion` to `1.5.0`. We may want to do the same. @bebound
